### PR TITLE
LPD-33263 Add ExternalReferenceCode to MBCategory (2nd try)

### PR DESCRIPTION
**Pull request by marcogalluzzi**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-33263

Add the ExternalReferenceCode to MBCategory entity

NOTE: added one [additional commit](https://github.com/liferay-content-management/liferay-portal/commit/0a5dd100920462f4317944e1f76e2ea4500aa759) to previous PR #5866 to simplify code according to what [requested by Brian](https://github.com/brianchandotcom/liferay-portal/pull/154326#issuecomment-2360958260)

# How am I fixing it?

Changing model, adding the ERC to ADD methods, create DELETE methods. Fixing all usages.
Also creating upgrade process.

# How can you verify that it works?

Run `gw tI --tests MBCategoryExternalReferenceCodeUpgradeProcessTest`

### CODE REVIEW

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBCategoryStagedModelDataHandler.java [hunk 1]
```diff
@@ -149,14 +149,21 @@ protected void doImportStagedModel(
 		MBCategory importedCategory = null;
 
 		if (portletDataContext.isDataStrategyMirror()) {
-			MBCategory existingCategory = fetchStagedModelByUuidAndGroupId(
-				category.getUuid(), portletDataContext.getScopeGroupId());
+			MBCategory existingCategory =
+				_mbCategoryLocalService.fetchMBCategoryByExternalReferenceCode(
```
Comment (Mikellorza) [inline modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBCategoryStagedModelDataHandler.java:153]: What is the reason to delete the log? I follow the structure of the classes and I didn't see a similar log in the api. Am i missing something? This log is useful when we are using staging and export/import process.

**Review by Mikellorza — CHANGES_REQUESTED** [review]

File: modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBCategoryStagedModelDataHandler.java (hunk 1 above)
Comment (marcogalluzzi) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBCategoryStagedModelDataHandler.java:153]: Well, so I don't fully understand Brian suggestion. Maybe he wants me to use get+try+catch ?

Comment (marcogalluzzi) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBCategoryStagedModelDataHandler.java:153]: BTW, I copied this log structure from another class.

Comment (Mikellorza) [reply modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBCategoryStagedModelDataHandler.java:153]: Yes, you have to use getMBCategoryByExternalReferenceCode and inside it there is a log similiar when we are calling to findByERC_G of the persistence and of corse, you will have to catch the exception

**Comment by Mikellorza** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**